rednotebook/gui/search.py

```diff
@@ -60,18 +60,21 @@ def search(self, search_text):
             else:
                 queries.append(part)
 
-        search_text = " ".join(queries)
-
-        # Highlight all occurences in the current day's text
-        self.main_window.highlight_text(search_text)
+        # Highlight all occurrences in the current day's text
+        self.main_window.highlight_text(queries)
 
         # Scroll to query.
-        if search_text:
-            GObject.idle_add(
-                self.main_window.day_text_field.scroll_to_text, search_text
-            )
-
-        self.main_window.search_tree_view.update_data(search_text, tags)
+        if queries:
+            # TODO: Decide where to scroll to?
+            # To scroll to the first match, we'd need to search for the first
+            # occurrence of each query, and scroll to the first one. But I am
+            # not sure if that is ideal performance-wise.
+            # Would it make sense to inspect the data returned by
+            # `self.journal.search` -- which will have information on what
+            # query matched fist, and it's index?
+            pass
```

Comment:
I think it's fine to use the old code for this. It should be fast enough. We can simply scroll to the first query that's not part of the day's date. For this we can pass `tags` and queries` to `scroll_to_text` and let it pick the first non-date query.

Reply:
> For this we can pass `tags` and `queries` to `scroll_to_text` and let it pick the first non-date query.

Hey, not sure what you meant by that: Do we need to change the implementation of `scroll_to_text` to accept a list type argument (where we'd pass `tags + queries`) instead of `text` (string type) -- Because, my understanding is that `it.forward_search` (which is used internally by `scroll_to_text`) only accepts string arguments.

Even if we do that, it won't be able to decide which query to pick automatically, right?

Reply:
Yes, we need to change `scroll_to_text`. It gets the information about the whole query and then loops over the info and chooses the first string that is not part of the day's date (e.g., "2020-02-28"). I think the query information can be passed as a list of strings (by concatenating tags and queries).

Reply:
Got it, thanks for the prompt reply!

I am trying to figure out if there's a way to access the current day's string, from `scroll_to_text` to no avail. I wanted to ask you -- is there an easy way to refer to the `Day` object from `search_to_text` method?

Reply:
Ah, nvm, I figured it -- it was as simple as calling `self.day`. 🤦